File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem33_MaximumDepthOfBinaryTreeTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem33_MaximumDepthOfBinaryTreeTest {

    private final Problem33_MaximumDepthOfBinaryTree test = new Problem33_MaximumDepthOfBinaryTree();

    @Test
    void maxDepth() {
        TreeNode root = new TreeNode(3);
        root.left = new TreeNode(9);
        root.right = new TreeNode(20);
        root.right.left = new TreeNode(15);
        root.right.right = new TreeNode(7);

        assertEquals(3, test.maxDepth(root));
    }
}

File: JavaScript/leetcode75/src/test/Problem33_MaximumDepthOfBinaryTree.test.js
const maxDepth = require("../main/Problem33_MaximumDepthOfBinaryTree");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return max depth", () => {
  let root = new TreeNode(3, 9, 20);
  root.left = new TreeNode(9, null, null);
  root.right = new TreeNode(20, 15, 7);

  expect(maxDepth(root)).toBe(3);
});
